deployment-templates/compute-engine/server/forseti-instance-server.py

```diff
@@ -203,7 +203,15 @@ def GenerateConfig(context):
 echo "$FORSETI_ENV" > $USER_HOME/forseti_env.sh
 
 USER=ubuntu
-(echo "{run_frequency} $FORSETI_HOME/setup/gcp/scripts/run_forseti.sh") | crontab -u $USER -
+
+# Use flock to prevent rerun of the same cron job when the previous job is still running.
+# If the lock file does not exist under the tmp directory, it will create the file and put a lock on top of the file.
+# When the previous cron job is not finished and the new one is trying to run, it will attempt to acquire the lock
+# to the lock file and fail because the file is already locked by the previous process.
+# The -n flag in flock will fail the process right away when the process is not able to acquire the lock so we won't
+# queue up the jobs.
+
+(echo "{run_frequency} /usr/bin/flock -n /tmp/forseti_cron_runner.lock $FORSETI_HOME/setup/gcp/scripts/run_forseti.sh") | crontab -u $USER -
```

Comment:
Does this resolve the scenario when the user-triggered forseti process is running, and it would be killed by the cron job restarting the server?

Reply:
No, it does not - the issue you mentioned would be a separate fix to have the server not restart when we pull in the newer configurations from the gcs bucket.

Reply:
I think there could / should be one solution that would apply to both scenarios, i.e. detect there is a running forseti process, whether cron-triggered or user-triggered, and should not be killed.  So, I like to hear more about how you would approach this.

Reply:
Okay, let's go with this as discussed offline for now.  But it would be ideal for the second cron job to run in parallel alongside the first one, rather than just abort the second one, so that any environment change can be picked up.  We really also need to address the server not restarting so that user-triggered process does not get killed.